internal/bminventory/inventory.go

```diff
@@ -4110,6 +4112,29 @@ func (b *bareMetalInventory) getHost(ctx context.Context, clusterId string, host
 	return &host, nil
 }
 
+func (b *bareMetalInventory) GetHostInternal(ctx context.Context, clusterId string, hostId string) (*common.Host, error) {
+	log := logutil.FromContext(ctx, b.log)
+	var host common.Host
+
+	if err := b.db.First(&host, "id = ? and cluster_id = ?", hostId, clusterId).Error; err != nil {
```

Comment:
suggesting to add a generic function to https://github.com/openshift/assisted-service/blob/master/internal/common/db.go 
or use `getHost` that is defined in the same file. 
In addition you can't use `StatusNotFound` without checking that this is actually the error `gorm.IsRecordNotFoundError(err)`

Reply:
Done